test/e2e/restart_changed_config/restart_changed_config_test.go

```diff
@@ -0,0 +1,176 @@
+	gomega.Ω(ns.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+	ns, err = kClient.CreateNamespace(test, nil)
+	gomega.Ω(err).NotTo(gomega.HaveOccurred())
+	gomega.Ω(ns.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	ginkgo.By("Tear down namespace: " + dev)
+	err := kClient.TearDownNamespace(dev)
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Tear down namespace: " + test)
+	err = kClient.TearDownNamespace(test)
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	// call the healthCheck api to check scheduler health
+	ginkgo.By("Check YuniKorn's health")
+	checks, err2 := yunikorn.GetFailedHealthChecks()
+	Ω(err2).NotTo(gomega.HaveOccurred())
+	Ω(checks).To(gomega.Equal(""), checks)
+
+	ginkgo.By("Restoring the old config maps")
+	var c, err1 = kClient.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
+		configmanager.DefaultYuniKornConfigMap)
+	Ω(err1).NotTo(gomega.HaveOccurred())
+	Ω(c).NotTo(gomega.BeNil())
+	c.Data = oldConfigMap.Data
+	var e, err3 = kClient.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
+	Ω(err3).NotTo(gomega.HaveOccurred())
+	Ω(e).NotTo(gomega.BeNil())
+})
+
+var _ = ginkgo.Describe("PodInRecoveryQueue", func() {
+	ginkgo.It("Pod_Restored_In_Recovery_Queue", func() {
+		ginkgo.By("Deploy 1st sleep pod to the dev namespace")
+		podConf, podErr := k8s.InitSleepPod(sleepPodConfigs)
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		podDev, err := kClient.CreatePod(podConf, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodRunning(dev, podDev.Name, 1*time.Minute)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Deploy 2nd sleep pod to the test namespace")
+		podConf, podErr = k8s.InitSleepPod(sleepPod2Configs)
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		var podTest *v1.Pod
+		podTest, err = kClient.CreatePod(podConf, test)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodRunning(test, podTest.Name, 1*time.Minute)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "fifo", func(sc *configs.SchedulerConfig) error {
+			// remove placement rules: only allow dev queue
+			sc.Partitions[0].PlacementRules = []configs.PlacementRule{{
+				Name:   types.Tag,
+				Value:  "namespace",
+				Create: false,
+			}}
+
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:   dev,
+				Parent: false,
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+
+		ginkgo.By("Check pod in the dev namespace")
+		var appsInfo *dao.ApplicationDAOInfo
+		appsInfo, err = restClient.GetAppInfo("default", "root."+dev, podDev.ObjectMeta.Labels["applicationId"])
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(appsInfo).NotTo(gomega.BeNil())
+
+		ginkgo.By("Check pod in the test namespace: recovery queue")
```

Comment:
`recovery queue` is a bit weird here since the pod is located at `root.testX`

Reply:
that is a copy paste error from the line 166 should not have a queue mentioned, will fix on commit